spacy/cli/_util.py

```diff
@@ -596,6 +598,8 @@ def walk_directory(path: Path, suffix: Optional[str] = None) -> List[Path]:
             continue
         elif path.is_dir():
             paths.extend(path.iterdir())
+        elif suffix == AUTO:
+            locs.append(path)
```

Comment:
If I'm reading this correctly, this is the only logic change here. 

I think that the intent when this was written was that `suffix` should be `None` when `converter` is `auto`. While it's not a problem now given the limited ways we use this function, I think it makes sense to reserve special behavior for `None` rather than for a specific keyword, which kind of feels like mixing behavior. 

In that case, we can get the desired behavior by calculating suffix before this is called in conversion like this:

```
suffix = converter if converter != "auto" else None
... = walk_directory(path, suffix)
``` 

If we change this, then this particular `elif` segment can just be removed.

Also, I mentioned it in the last PR, but I don't think that the name of this function makes it clear what it's actually doing, so it would be good to add a docstring. Something like:

> Given a directory and a suffix, recursively find all files matching the suffix. Directories or files with names beginning with a `.` are ignored, but hidden flags on filesystems are not checked.

Reply:
>I think that the intent when this was written was that `suffix` should be `None` when `converter` is `auto`. While it's not a problem now given the limited ways we use this function, I think it makes sense to reserve special behavior for None rather than for a specific keyword, which kind of feels like mixing behavior.

That makes sense, then we can keep the "auto" stuff limited to the `convert` functionality only.

Reply:
This required a little more additional work, as the `verify_cli_args` function is impacted too. I moved all functionality of converting "auto" to the right suffix + warnings to the `_get_converter` function which now should be called before `verify_cli_args`. I also added more specific CLI tests.